app/code/Bulbulatory/Recomendations/Ui/Component/Listing/RecommendationsTableDataProvider.php

```diff
@@ -0,0 +1,23 @@
+<?php
+namespace Bulbulatory\Recomendations\Ui\Component\Listing;
+
+class RecommendationsTableDataProvider extends \Magento\Framework\View\Element\UiComponent\DataProvider\SearchResult
+{
+    protected function _initSelect()
+    {
+        parent::_initSelect();
+        $this->getSelect()->joinLeft(
+            [
+                'secondTable' => $this->getTable('customer_entity')
```

Comment:
Lepiej dawać nazwę, która bardziej odpowiada łączonej tabeli - w tym wypadku np. `customer` albo coś podobnego, co wskazuje że jest to tabela z użytkownikami.